apiclient/src/lib.rs

```diff
@@ -60,43 +64,62 @@ impl std::fmt::Display for DomainOrAddress {
 #[derive(Clone)]
 pub struct ApiClient {
     client: Client,
-    url: String,
+    port: u16,
+    hostname: String,
+    tls_enabled: bool,
 }
 
 impl ApiClient {
     /// Creates a new API client that connects to the given base URL.
     ///
     /// # Arguments
-    /// base_url - The base URL of the server.
-    /// transport_encryption - Whether transport encryption is enabled or not.
+    /// url - The base URL or hostname:port tuple of the server. If the URL
+    /// starts with `https`, TLS will be used. If the URL or hostname:port tuple
+    /// includes a port, that port will be used, otherwise, if TLS is enabled,
+    /// the default port is 443, if TLS is disabled, the default port is 8000.
     ///
     /// # Returns
     /// A new [`ApiClient`].
     pub fn initialize(domain: impl ToString) -> Result<Self, ApiClientInitError> {
+        let mut domain_string = domain.to_string();
+        // If the url doesn't start with http, we assume it's a hostname/port
+        // combination and we prefix it with "http://"
+        if !domain_string.starts_with("http") {
+            domain_string.insert_str(0, "http://");
+        }
+        let url = Url::parse(domain_string.as_str())
+            .map_err(|_| ApiClientInitError::UrlParsingError(domain_string.clone()))?;
+        let tls_enabled = url.scheme() == "https";
+        let port = url
+            .port()
+            .unwrap_or_else(|| if tls_enabled { 443 } else { 8000 });
```

Comment:
Why not port 80? We can still specify whatever we want for the tests.

Reply:
We could do 80, but applications require elevated privileges to open ports below 1024, so 8000 makes testing easier, where we typically don't have those privileges. Come to think of it, we probably don't want to use 443 here, either.